wandb/sdk/internal/sender.py

```diff
@@ -105,9 +108,12 @@ def __init__(
         # State added when run_exit is complete
         self._exit_result = None
 
-        self._api = internal_api.Api(default_settings=settings)
+        self._api = internal_api.Api(default_settings=settings, retry_callback=self.retry_callback)
         self._api_settings = dict()
 
+        # queue filled by retry_callback
+        self._retry_q: "Queue[HttpResponse]" = multiprocessing.Queue()
```

Comment:
nit: This probably isn't worth changing, but you could technically use a regular queue here.